raft/log.go

```diff
@@ -1021,9 +1026,11 @@ func (l *Log) candidateLoop(closing <-chan struct{}) State {
 			l.lock()
 			l.mustSetTermIfHigher(hb.term)
 			if hb.term >= l.term {
+				l.tracef("candidateLoop: new leader: old=%d new=%d", l.leaderID, hb.leaderID)
 				l.leaderID = hb.leaderID
 			}
 			l.unlock()
+			return Follower
```

Comment:
In other words, by putting it inside the `if` block, the code is fully compliant with the Raft paper. By putting it outside, it may not be.